packages/rx_bloc_cli/mason_templates/bricks/lib_auth_apple/__brick__/lib/lib_auth_apple/blocs/apple_login_bloc.dart

```diff
@@ -0,0 +1,56 @@
+{{> licence.dart }}
+
+import 'package:rx_bloc/rx_bloc.dart';
+import 'package:rxdart/rxdart.dart';
+
+import '../../base/common_blocs/coordinator_bloc.dart';
+import '../../base/extensions/error_model_extensions.dart';
+import '../../base/models/errors/error_model.dart';
+import '../../lib_auth/services/user_account_service.dart';
+
+part 'apple_login_bloc.rxb.g.dart';
+
+/// A contract class containing all events of the LoginWithAppleBloC.
+abstract class AppleLoginBlocEvents {
+  /// Initiate session with user's Apple credentials
+  void loginWithApple();
+}
+
+/// A contract class containing all states of the LoginWithAppleBloC.
+abstract class AppleLoginBlocStates {
+  /// The loading state
+  Stream<bool> get isLoading;
+
+  /// The error state
+  Stream<ErrorModel> get errors;
+
+  /// State indicating whether the user is logged in successfully
+  ConnectableStream<bool> get loggedIn;
+}
+
+@RxBloc()
+class AppleLoginBloc extends $AppleLoginBloc {
+  AppleLoginBloc(this._userAccountService, this._coordinatorBloc) {
+    loggedIn.connect().addTo(_compositeSubscription);
+  }
+
+  final UserAccountService _userAccountService;
+  final CoordinatorBlocType _coordinatorBloc;
+
+  @override
+  Stream<ErrorModel> _mapToErrorsState() => errorState.mapToErrorModel();
+
+  @override
+  Stream<bool> _mapToIsLoadingState() => loadingState;
+
+  @override
+  ConnectableStream<bool> _mapToLoggedInState() => _$loginWithAppleEvent
+      .throttleTime(const Duration(seconds: 1))
+      .exhaustMap((_) => _userAccountService.socialLogin().asResultStream())
```

Comment:
If you want to transform the type of the stream you can use the `.then` operator such as
_userAccountService.socialLogin().then((_) => true).asResultStream()

Reply:
Thank you! Will use it.